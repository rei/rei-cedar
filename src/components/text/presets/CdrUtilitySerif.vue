<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import { baseTextProps } from '../../../types/interfaces';

defineOptions({
  name: 'CdrUtilitySerif',
});

interface utilitySerifTextProps extends baseTextProps {
   /** 
   * Sets the type scale
   * @type scaleValue
   * @values -1,0,1,2,3
   */
  scale?: '-1'|'0'|'1'|'2'|'3',
  /** Toggles the strong variant */
  strong?: boolean,
}

const props = withDefaults(defineProps<utilitySerifTextProps>(), {
  tag: 'p',
  scale: '1',
  strong: false,
});

const typeProperties = computed(() => {
  return {
    '--cdr-utility-serif-font-size': `var(--cdr-type-scale-${props.scale})`,
    '--cdr-utility-serif-line-height': 
    `var(--cdr-line-height-ratio-utility-${props.scale})`,
    '--cdr-utility-serif-font-weight': props.strong ? '600' : '400',

  }
});

const baseClass = 'cdr-utility-serif';
const style = useCssModule();
</script>

<template>
  <component
    :is="tag"
    :class="style[baseClass]"
    :style="typeProperties"
  >
    <slot />
  </component>
</template>

<style module src="./styles/CdrUtilitySerif.module.scss" lang="scss">
</style>  