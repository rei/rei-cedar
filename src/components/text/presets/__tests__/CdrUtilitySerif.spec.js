import { mount } from '../../../../../test/vue-jest-style-workaround.js';
import CdrUtilitySerif from '../CdrUtilitySerif.vue';


describe('CdrUtilitySerif', () => {
  describe('default configuration', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrUtilitySerif, {
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('passing a scale', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrUtilitySerif, {
        props: { scale: '-1' },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('setting to strong variant', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrUtilitySerif, {
        props: { strong: true },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});
