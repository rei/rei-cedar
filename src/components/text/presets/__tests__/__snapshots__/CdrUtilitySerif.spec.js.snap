// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrUtilitySerif > default configuration > renders correctly 1`] = `
<p
  class="cdr-utility-serif"
  style="--cdr-utility-serif-font-size: var(--cdr-type-scale-1); --cdr-utility-serif-line-height: calc(var(--cdr-type-scale-1) * var(--cdr-utility-serif-line-height-ratio)); --cdr-utility-serif-font-weight: 400;"
>
  
  paragraph
  
</p>
`;

exports[`CdrUtilitySerif > passing a scale > renders correctly 1`] = `
<p
  class="cdr-utility-serif"
  style="--cdr-utility-serif-font-size: var(--cdr-type-scale--1); --cdr-utility-serif-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-utility-serif-line-height-ratio)); --cdr-utility-serif-font-weight: 400;"
>
  
  paragraph
  
</p>
`;

exports[`CdrUtilitySerif > setting to strong variant > renders correctly 1`] = `
<p
  class="cdr-utility-serif"
  style="--cdr-utility-serif-font-size: var(--cdr-type-scale-1); --cdr-utility-serif-line-height: calc(var(--cdr-type-scale-1) * var(--cdr-utility-serif-line-height-ratio)); --cdr-utility-serif-font-weight: 600;"
>
  
  paragraph
  
</p>
`;
